src/ui/inspector/PlotElementsPool.vue

```diff
@@ -0,0 +1,316 @@
+                }
+            }
+        },
+        unlistenComposition() {
+            if (this.compositionUnlistener) {
+                this.compositionUnlistener();
+            }
+        },
+        registerCompositionListeners() {
+            this.composition.on('add', this.addElement);
+            this.composition.on('remove', this.removeElement);
+            this.composition.on('reorder', this.reorderElements);
+
+            this.compositionUnlistener = () => {
+                this.composition.off('add', this.addElement);
+                this.composition.off('remove', this.removeElement);
+                this.composition.off('reorder', this.reorderElements);
+                delete this.compositionUnlistener;
+            };
+        },
+        setYAxisIds() {
+            const configId = this.openmct.objects.makeKeyString(this.parentObject.identifier);
+            this.config = configStore.get(configId);
+            this.yAxes.push({
+                id: this.config.yAxis.id,
+                elements: this.parentObject.configuration.series.filter(
+                    series => series.yAxisId === this.config.yAxis.id
+                )
+            });
+            if (this.config.additionalYAxes) {
+                this.config.additionalYAxes.forEach(yAxis => {
+                    this.yAxes.push({
+                        id: yAxis.id,
+                        elements: this.parentObject.configuration.series.filter(
+                            series => series.yAxisId === yAxis.id
+                        )
+                    });
+                });
+            }
+        },
+        addElement(element) {
+            // Get the index of the corresponding element in the series list
+            const index = this.parentObject.configuration.series.findIndex(
+                series => this.openmct.objects.areIdsEqual(series.identifier, element.identifier)
+            );
+            let yAxisId = this.parentObject.configuration.series[index].yAxisId;
+            if (yAxisId === undefined) {
+                yAxisId = Y_AXIS_1;
+
+                // Insert the element at the end of the YAxis1 bucket
+                let insertIndex = 0;
+                while (insertIndex < this.yAxes.length && this.yAxes[insertIndex].id === yAxisId) {
+                    insertIndex++;
+                }
+
+                this.composition.reorder(index, insertIndex + 1);
+            }
+
+            const keyString = this.openmct.objects.makeKeyString(element.identifier);
+
+            // Store the element in the cache and set its yAxisId
+            this.elementsCache[keyString] = JSON.parse(JSON.stringify(element));
+            if (this.elementsCache[keyString].yAxisId !== yAxisId) {
+                this.elementsCache[keyString].yAxisId = yAxisId;
+                // Mutate the YAxisId on the domainObject itself
+                this.mutateYAxisId(element, yAxisId);
```

Comment:
We need to test that mutateYAxisId is not called if a saved plot is loaded.

Reply:
Added in some logic to only mutate the domain object if the yAxisId has changed